### PR TITLE
feat(connector-iroha2): sending transactions signed on the client-side

### PR DESCRIPTION
**Pull request by outSH**
This is improvement to initial Iroha V2 PR. Please review only the last commit.

- Add new endpoint `generate-transaction`, to create unsigned transactions
  that can be signed on the client side.
- Add a function to iroha2-connector package to help signing iroha transactions
  on the client (BLP) side.
- Extend transact endpoint to accept signed transaction as an argument as well.
- Add new test suite to check features implemented in this PR (i.e. signing on the client side).

Relates to #2077

Depends on #2153

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

### CODE REVIEW

File: packages/cactus-plugin-ledger-connector-iroha2/README.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+- [Acknowledgments](#acknowledgments)
+
+## Remarks
+
+- Docker support is not implemented yet.
+- There is no official Iroha V2 release yet. API and connector behavior can change before stable release.
+- Query pagination is not supported yet. Querying large datasets (the ones with `All`) can be catastrophic. Pagination is not implemented in upstream javascript iroha sdk yet.
+- There's no way to check transaction status that we send. We're waiting for upstream change, see: https://github.com/hyperledger/iroha-javascript/issues/117
+- Connector can't be used with cactus module system yet. See: https://github.com/hyperledger/iroha-javascript/issues/111
+- Signing transactions on the client side is not supported yet, we're waiting for upstream change, see: https://github.com/hyperledger/iroha-javascript/issues/113
```
Comment (izuru0) [inline packages/cactus-plugin-ledger-connector-iroha2/README.md:23]: This description seems old

Comment (outSH) [reply packages/cactus-plugin-ledger-connector-iroha2/README.md:23]: Please review only the last commit here, base iroha2 connector commit is not updated with original PR. Will do the rebase after irohav2 PR is merged.

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Review by petermetz — CHANGES_REQUESTED** [review]
@outSH Please resolve the merge conflicts! (also squash+rebase)

**Pull request closed without merging**